gorp_go18_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2012 James Cooper. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+// Package gorp provides a simple way to marshal Go structs to and from
+// SQL databases.  It uses the database/sql package, and should work with any
+// compliant database/sql driver.
+//
+// Source code and project home:
+// https://github.com/go-gorp/gorp
+//
+
+// +build go1.8
+
+package gorp_test
+
+import (
+	"testing"
+	"time"
+	"context"
+	"github.com/stretchr/testify/assert"
+)
+
+// Drivers that don't support cancellation.
+var unsupportedDrivers map[string]bool = map[string]bool{
+	"mymysql":true,
+}
+
+
+func TestWithNotCanceledContext(t *testing.T) {
+	dbmap := initDbMap()
+	defer dropAndClose(dbmap)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+	defer cancel()
+
+	withCtx := dbmap.WithContext(ctx)
+
+	_, err := withCtx.Exec("SELECT 1")
+	assert.Nil(t, err)
+}
+
+func TestWithCanceledContext(t *testing.T) {
+	dialect, driver := dialectAndDriver()
+	if unsupportedDrivers[driver] {
+		t.Skip("Cancellation is not yet supported by all drivers. Not known to be unsupported in %s.", driver)
```

Comment:
I think this should read `Known to be unsupported` or `Not known to be supported`, rather than `Not known to be unsupported`.